kolena/initialize.py

```diff
@@ -11,23 +11,28 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import netrc
+import os
 import warnings
 from typing import Any
 from typing import Dict
 from typing import Optional
+from urllib.parse import urlparse
 
 import pandas as pd
 
 from kolena._utils import log
 from kolena._utils import state
+from kolena._utils.consts import KOLENA_TOKEN_ENV
 from kolena._utils.endpoints import get_platform_url
 from kolena._utils.instrumentation import upload_log
 from kolena._utils.state import _client_state
 from kolena.errors import InputValidationError
+from kolena.errors import MissingTokenError
 
 
 def initialize(
-    api_token: str,
+    api_token: Optional[str] = None,
     *args: Any,
```

Comment:
🤔 could it break anything to have the `*args` positional argument catchall come after an optional arg?

Reply:
I don't believe anything can be broken since the deprecated signature is `initialize(entity, api_token)` and we check `if len(args) == 1` to overwrite the entity sitting in api_token param with `args[0]`.
An additional change I considered was moving the `api_token` to be keyword only so it is in line with `verbose` and `proxy`. Decided against this to keep the PR trimmed down, but I do think this is the right thing to do going forward.